tests/unit/type/list.tcl

```diff
@@ -276,6 +315,36 @@ start_server {
         assert_equal {foo} [r lrange target 0 -1]
     }
 
+    test "BRPOPRPUSH with zero timeout should block indefinitely" {
+        set rd [redis_deferring_client]
+        r del blist target
+        $rd brpoprpush blist target 0
+        after 1000
```

Comment:
for these 4 tests, i'd replace the `after 1000` with a `wait_for_condition` for `[s blocked_clients] == 1`
that would make all 4 run after than the original 1 too.
then i'd also suggest to put something in the target list before the operation, and then check that by using the right kind of pop, we get the right item.
this (together with the above/previous non-blocking test), makes sure that both code paths, push to the right place.

Reply:
Done. Instead of checking the result with a pop, I just modified the assert adding the existing element in the right place.